From 4dfe7f9e702c85243a71e4de267a13e434b6d6c2 Mon Sep 17 00:00:00 2001
From: Jean Delvare <jdelvare@suse.de>
Date: Fri, 20 Jan 2023 12:56:08 +0100
Subject: [PATCH] test: Fix a race condition

The test suite does not differentiate between stdout and stderr. When
messages are printed to both, the order in which they will reach us
is apparently not guaranteed. Ideally this would be deterministic, but
until then, explicitly test stdout and stderr separately in the test
case itself. Otherwise the test suite fails randomly, which is a pain
for distribution package maintainers.

This fixes bug #63651 reported by Ross Burton:
https://savannah.nongnu.org/bugs/index.php?63651

Signed-off-by: Jean Delvare <jdelvare@suse.de>
---
 test/faildiff.test | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/test/faildiff.test b/test/faildiff.test
index 5afb8e3..0444c15 100644
--- a/test/faildiff.test
+++ b/test/faildiff.test
@@ -27,8 +27,9 @@ What happens on binary files?
 	> File test.bin added to patch %{P}test.diff
 
 	$ printf "\\003\\000\\001" > test.bin
-	$ quilt diff -pab --no-index
+	$ quilt diff -pab --no-index 2>/dev/null
 	>~ (Files|Binary files) a/test\.bin and b/test\.bin differ
+	$ quilt diff -pab --no-index >/dev/null
 	> Diff failed on file 'test.bin', aborting
 	$ echo %{?}
 	> 1
-- 
2.39.2

